backend/stakepoold/server.go

```diff
@@ -371,43 +813,177 @@ func (ctx *appContext) vote(wg *sync.WaitGroup, blockHash *chainhash.Hash, block
 
 	// Ask wallet to transmit raw transaction.
 	startSend := time.Now()
-	_, w.err = ctx.nodeConnection.SendRawTransaction(newTx, false)
+	tx, err := ctx.nodeConnection.SendRawTransaction(newTx, false)
+	if err != nil {
+		w.err = err
+	} else {
+		w.txid = tx
+	}
 	w.sendDuration = time.Since(startSend)
 }
 
+func (ctx *appContext) processNewTickets(nt NewTicketsForBlock) {
+	start := time.Now()
+
+	// We use pointer because it is the fastest accessor.
+	newtickets := make([]*ticketMetadata, 0, len(nt.newTickets))
+
+	var wg sync.WaitGroup // wait group for go routine exits
+
+	ctx.mtx.RLock()
+	for _, tickethash := range nt.newTickets {
+		n := &ticketMetadata{
+			blockHeight: nt.blockHeight,
+			ticket:      tickethash,
+			ticketType:  ticketTypeNew,
+		}
+		newtickets = append(newtickets, n)
+
+		wg.Add(1)
+		go ctx.getticket(&wg, n)
+	}
+	ctx.mtx.RUnlock()
+
+	wg.Wait()
+
+	addtickets := make(map[chainhash.Hash]string)
+
+	for _, n := range newtickets {
+		if n.err != nil || n.msa == "" {
+			// most likely can't look up the transaction because it's
+			// not in our wallet because it doesn't belong to us
+			continue
+		}
+
+		addtickets[*n.ticket] = n.msa
+	}
+
+	ticketCountNew := 0
+	ticketCountOld := 0
+
+	ctx.mtx.Lock()
+	ticketCountOld = len(ctx.liveTicketsMSA)
+	for ticket, msa := range addtickets {
+		ctx.liveTicketsMSA[ticket] = msa
+	}
+	ticketCountNew = len(ctx.liveTicketsMSA)
+	ctx.mtx.Unlock()
+	atomic.StoreUint64(&ctx.stats.TicketsLiveCount, uint64(ticketCountNew))
```

Comment:
It's gonna be pretty difficult to keep the stats fields consistent with each other.  Either it doesn't matter, or they all need to be set at once.  In this function just adding a mutex to Stats and locking it might work, but since there are a few other spots where `ctx.stats.TicketsLiveCount` is potentially updated, I'm not sure what a solution to that might be.